### PR TITLE
Fix errors from release management

### PR DESCRIPTION
**Pull request by acidmoose**
* Use "mvn package" as dry run deployment step until gpg
keys are available, so don't accidentally deploy non-signed artifacts.
* Remove duplicate Guava version from jsonlogging
* Remove unused version properties from jsonlogging
* Remove workflows from CircleCI. Stick with sequential flow until needed.
* Add plugin management of missing version for deploy plugin.
* Add version to gpg plugin.
* Add gpg environment variables for future gpg signing.
* Fix javadoc warning for SecretHandler[] link.
* Add gpg signing based on environment variables:
    - GPG_KEY - base64 encoded private key to import
    - GPG_KEY_NAME - the name of the key imported
    - GPG_KEY_PASSPHRASE - the password protecting the key
    - OSSRH_JIRA_USERNAME - username on sonatype maven central with access to deploy
    - OSSRH_JIRA_PASSWORD - password on sonatype maven central with access to deploy

### CODE REVIEW

**Comment by acidmoose** [comment]
* Use "mvn package" as dry run deployment step until gpg
keys are available, so  don't accidentally deploy non-signed artifacts. 
* Remove duplicate Guava version from jsonlogging
* Remove unused version properties from jsonlogging
* Remove workflows from CircleCI. Stick with sequential flow until needed.
* Add plugin management of missing version for deploy plugin.
* Add version to gpg plugin.
* Add gpg environment variables for future gpg signing.
* Fix javadoc warning for SecretHandler[] link.

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,63 +23,23 @@ jobs:
           # fallback to using the latest cache if no exact match is found
           - v1-dependencies-
 
-      - run: mvn package dependency:go-offline
+      #- run: mvn package dependency:go-offline
```
Comment (kbarek) [inline .circleci/config.yml]: Remove commented line?

File: .circleci/config.yml [hunk 2]
```diff
@@ -23,63 +23,23 @@ jobs:
...
 
       - save_cache:
           paths:
             - ~/.m2
           key: v1-dependencies-{{ checksum "pom.xml" }}
 
-      # run tests!
```
Comment (kbarek) [inline .circleci/config.yml]: Some comments start with capital letter and some don't. If you want nitpick mode I'd say be consistent :)

File: .circleci/config.yml [hunk 3]
```diff
@@ -23,63 +23,23 @@ jobs:
...
-            - ~/.m2
-          key: v1-dependencies-{{ checksum "pom.xml" }}
-
-      # run tests!
-      - run: mvn integration-test
-
-      - command: |
-          if [ "${CIRCLE_BRANCH}" == "master" ]; then
-            mvn -s .circleci/settings.xml -DskipTests -DperformRelease deploy
+      #- run: mvn integration-test
```
Comment (kbarek) [inline .circleci/config.yml]: Remove commented line?

File: .circleci/config.yml [hunk 4]
```diff
@@ -23,63 +23,23 @@ jobs:
...
+
+            #mvn -s .circleci/settings.xml -DskipTests -DperformRelease verify;
```
Comment (kbarek) [inline .circleci/config.yml]: And here

File: .circleci/config.yml [hunk 5]
```diff
@@ -23,63 +23,23 @@ jobs:
...
+      # deploy if release branch
+      - run: |
+          if [[ $CIRCLE_BRANCH = 'release' ]]; then
+            echo $GPG_KEY | base64 --decode > signing-key
+            gpg --pinentry-mode loopback --passphrase $GPG_PASSPHRASE --import signing-key
+            shred signing-key
+            mvn -s .circleci/settings.xml -DskipTests -DperformRelease verify
+            ls target
```
Comment (kbarek) [inline .circleci/config.yml]: Is this necessary outside of testing?

File: .circleci/config.yml [hunk 6]
```diff
@@ -23,63 +23,23 @@ jobs:
...
-      - command: |
-          if [ "${CIRCLE_BRANCH}" == "master" ]; then
-            mvn -s .circleci/settings.xml -DskipTests -DperformRelease deploy
+      #- run: mvn integration-test
+
+            #mvn -s .circleci/settings.xml -DskipTests -DperformRelease verify;
+      # deploy if release branch
+      - run: |
+          if [[ $CIRCLE_BRANCH = 'release' ]]; then
+            echo $GPG_KEY | base64 --decode > signing-key
```
Comment (neumayer) [inline .circleci/config.yml]: I thought we didn't need this any more?

File: .circleci/config.yml [hunk 7]
```diff
@@ -23,63 +23,23 @@ jobs:
...
-          filters:
-            branches:
-              only: release
-      - build-and-deploy:
-          filters:
-            branches:
-              only: release
-          requires:
-            - hold
+          echo "done"
```
Comment (neumayer) [inline .circleci/config.yml]: Also this is not strictly necessary.

File: jsonlogging/pom.xml [hunk 8]
```diff
@@ -80,11 +78,6 @@
             <version>${joda-time.version}</version>
         </dependency>
```
Comment (neumayer) [inline jsonlogging/pom.xml:79]: This is not too much related, is it? Extract to separate pr?

File: .circleci/config.yml (hunk 1 above)
Comment (acidmoose) [reply .circleci/config.yml]: Removed during testing. Added again.

File: .circleci/config.yml (hunk 3 above)
Comment (acidmoose) [reply .circleci/config.yml]: Removed during testing. Added again.

File: .circleci/config.yml (hunk 4 above)
Comment (acidmoose) [reply .circleci/config.yml]: Removed

File: .circleci/config.yml (hunk 6 above)
Comment (acidmoose) [reply .circleci/config.yml]: We need to make a file out of it so that I can import it into the agent.

File: .circleci/config.yml (hunk 5 above)
Comment (acidmoose) [reply .circleci/config.yml]: Removed

File: .circleci/config.yml (hunk 7 above)
Comment (acidmoose) [reply .circleci/config.yml]: Removed. Used for testing.

File: jsonlogging/pom.xml (hunk 8 above)
Comment (acidmoose) [reply jsonlogging/pom.xml:79]: If I have to... This would be 9 PRs I guess. I just wanted to get it over with.

**Review by neumayer — APPROVED** [review]

File: pom.xml [hunk 9]
```diff
@@ -225,13 +234,21 @@
...
+                        <version>1.6</version>
                         <executions>
                             <execution>
                                 <id>sign-artifacts</id>
                                 <phase>verify</phase>
                                 <goals>
                                     <goal>sign</goal>
                                 </goals>
+                                <configuration>
+                                <!-- This is necessary for gpg to not try to use the pinentry programs -->
```
Comment (neumayer) [inline pom.xml:246]: Remove the "s?"

Comment (acidmoose) [reply pom.xml:246]: It's a collection of programs :)
https://gnupg.org/software/pinentry/index.html
